### PR TITLE
[JENKINS-21600], [JENKINS-60163] Finish Paging Implementation for Job and System config history

### PR DESCRIPTION
**Pull request by RobinRSchulz**
- Finishes the paging implementation for jobs and system configs (except with using filter=created in the overview page).

- Fixes an unrelated bug which caused an exception to be thrown on Jenkins startup, sometimes. (SYSTEM user was unnecessarily tried to be gotten)

### CODE REVIEW

File: src/main/java/hudson/plugins/jobConfigHistory/ComputerConfigHistoryAction.java [hunk 1]
```diff
@@ -124,6 +124,12 @@ public boolean hasConfigurePermission() {
 		return getAccessControlledObject().hasPermission(Computer.CONFIGURE);
 	}
 
+	@Override
+	public int getRevisionAmount() {
+		LOG.log(Level.WARNING, "Paging is not implemented for agents, yet!");
```
Comment (stefanbrausch) [inline src/main/java/hudson/plugins/jobConfigHistory/ComputerConfigHistoryAction.java]: Really a warning or only info?

Comment (RobinRSchulz) [reply src/main/java/hudson/plugins/jobConfigHistory/ComputerConfigHistoryAction.java]: done

File: src/main/java/hudson/plugins/jobConfigHistory/JobConfigHistoryRootAction.java [hunk 2]
```diff
@@ -208,6 +369,72 @@ public final String getIconFileName() {
...
+
+		final List configs = HistoryDescrToConfigInfo.convert(name, true, cuttedHistoryDescrs, false);
+		Collections.sort(configs, ParsedDateComparator.DESCENDING);
+		return configs;
+	}
+
+	@Override
+	public int getRevisionAmount() {
+		final String filter = getRequestParameter("filter");
+		final String historyRequestURI = "/jenkins/" + JobConfigHistoryConsts.URLNAME + "/history";
```
Comment (RobinRSchulz) [inline src/main/java/hudson/plugins/jobConfigHistory/JobConfigHistoryRootAction.java]: change equals to endswith and remove "/jenkins/"

Comment (RobinRSchulz) [reply src/main/java/hudson/plugins/jobConfigHistory/JobConfigHistoryRootAction.java]: Done.

**Pull request merged**